test/app.js

```diff
@@ -21,97 +21,135 @@ const app = rewire('../app.js');
 
 const validateArgs = app.__get__('validateArgs');
 
+function assertErrorMessage(msg, done) {
+  return function(err, data) {
+    try {
+      assert(err.message, 'There should be an error message');
+      assert(err.message.indexOf(msg) >= 0);
+      assert(data === null || data === undefined, 'There should only be an error.');
+      done();
+    } catch (e) {
+      done(e);
+    }
+  };
+}
+
+function assertNoError(done) {
+  return function(err, data) {
+    try {
+      assert(err === null, 'There should be no error message.');
+      done();
```

Comment:
What about checking the `data` if not null?

Reply:
I initially tried, but on closer inspection the data is always `null` at the validation stage (the callback only gets called with data when the backup completes). So it's not a valid check.